src/main/java/com/vertexvis/model/serialization/AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullTypeAdapter.java

```diff
@@ -0,0 +1,157 @@
+package com.vertexvis.model.serialization;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import com.vertexvis.model.AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullType;
+import com.vertexvis.model.MetadataDateType;
+import com.vertexvis.model.MetadataFloatType;
+import com.vertexvis.model.MetadataLongType;
+import com.vertexvis.model.MetadataNullType;
+import com.vertexvis.model.MetadataStringType;
+
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+public class AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullTypeAdapter
+        extends
+        TypeAdapter<AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullType> {
+
+    private final Supplier<Gson> gsonSupplier;
+
+    public AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullTypeAdapter(
+            Supplier<Gson> gsonSupplier) {
+        this.gsonSupplier = gsonSupplier;
+    }
+
+    @Override
+    public void write(JsonWriter out,
+                      AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullType value)
+            throws IOException {
+        out.jsonValue(gsonSupplier.get().toJson(value.getValue()));
+    }
+
+    @Override
+    public AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullType read(
+            JsonReader in) throws IOException {
```

Comment:
Deserialization of the metadata type.  Made more complicated by the `anyOf` qualifier in the schema.  The `type` field always wins in the case of ambiguity.